src/fidesops/service/connectors/query_config.py

```diff
@@ -553,13 +554,8 @@ def transform_query_pairs(pairs: Dict[str, Any]) -> Dict[str, Any]:
         if input_data:
             filtered_data: Dict[str, Any] = self.typed_filtered_values(input_data)
             if filtered_data:
-                field_list = {
-                    field_path.string_path: 1
-                    for field_path, field in self.field_map().items()
-                }
```

Comment:
The `field_list` portion here is used to generate the projection of the mongo query which specifies which fields we want from mongo.  `self.field_map` flattens every field and every level of a nested field.

The projection being generated was something like this: notice that `preferences.notifications` and `preferences.notifications.security_emails` are both included. This would cause mongo to throw errors about path collisions. 

```
db.getCollection('customer_details').find({"customer_id": 1}, {'_id': 1, 'preferences.notifications.security_emails': 1, 'preferences.notifications.informational_emails': 1, 'preferences.notifications': 1})
```

Below, I generate the `field_list` using top-level fields instead.  I'm also moving the definition of `field_list` closer to where it's being used.